spec/integration/type/tidy_spec.rb

```diff
@@ -11,6 +11,28 @@
     Puppet::Util::Storage.stubs(:store)
   end
 
+  it "should be able to recursively remove directories" do
+    dir = tmpfile("tidy_testing")
+    foo = File.join(dir, "foo")
+    bar = File.join(foo, "bar")
+
+    Dir.mkdir(dir)
+    Dir.mkdir(foo)
+    Dir.mkdir(bar)
+
+    tidy = Puppet::Type.type(:tidy).new :path => dir, :recurse => true, :rmdirs => true
+
+    catalog = Puppet::Resource::Catalog.new
+    catalog.add_resource(tidy)
+    # avoid crude failures because of nil resources that result
+    # from implicit containment and lacking containers
+    catalog.stubs(:container_of).returns tidy
+
+    catalog.apply
+
+    expect(Puppet::FileSystem.directory?(dir)).to be_falsey
```

Comment:
It definitely fails for me if I check out the previous version of the code. Are you sure you got the updated test file alongside the old version of the `additional_resource_generator`?

This gist shows getting the old version of the code, failing test, grabbing code from this PR, and then test passing: https://gist.github.com/branan/77a266161f92f82ff43b